cmake: 3.22.3 -> 3.23.2

AndersonTorres opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

New version and a cleanup of the expression.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

bjornfor commented:
>New version and a cleanup of the expression.

Can you split it in two commits, to ease review / git bisect?

AndersonTorres commented:
Done!

AndersonTorres committed:
cmake: reformat expression

AndersonTorres committed:
cmake: use a list of suitable uiToolkits instead of Boolean values

Also, aborts when the list contains anything besides the acceptable options.

AndersonTorres committed:
cmake: 3.22.3 -> 3.23.2

AndersonTorres committed:
cmake: add myself as maintainer

AndersonTorres commented:
Why that strange error?

AndersonTorres committed:
cmake: use callPackage instead of libsForQt5.callPackage

Following the trend of enhanced composability, and further because it is unusual
a build tool using libsForQt5 as scope.

bjornfor commented on pkgs/development/tools/build-managers/cmake/default.nix:
```diff
@@ -20,18 +20,21 @@
 }:
 
+let
+  cursesUI = lib.elem "ncurses" uiToolkits;
```
I'm not sure I agree that `uiToolkits` list instead of booleans is a cleaner interface, but no strong feelings. But, more importantly, it seems the current implementation doesn't warn the user of typos, which I think is bad UX. Please check uidToolkits for valid elements?

AndersonTorres replied:
My "strong feeling", to be honest, is the lack of consistency: `useNcurses` and `withQt` are completely differentm and it betrays completely the idea they should convey.

Ok, I will implement this as an assertion.

bjornfor commented:
> cmake: use callPackage instead of libsForQt5.callPackage

Doensn't that increase the risk of problems like https://github.com/NixOS/nixpkgs/pull/171380? Or perhaps it's equally bad, so using one or the other doesn't change the risk of library clashes?

AndersonTorres commented:
> Doensn't that increase the risk of problems like #171380? Or perhaps it's equally bad, so using one or the other doesn't change the risk of library clashes?

Honestly I don't know. 
I believe that it would make no difference here, because QT is employed uniquely to provide a fancy graphical UI. Indeed, only qtbase and the qt wrapper are used.
Cmake is neither a QT package nor rely on it to its main function (aka generate build files).

AndersonTorres committed:
cmake: rename patches

Because it is easier to look at them as a block, not polluting the directory
listing so much.

bobby285271 commented:
Looks like `005-remove-systemconfiguration-dep.diff` does not apply.

AndersonTorres commented:
> Looks like `005-remove-systemconfiguration-dep.diff` does not apply.

Is this a case for removal of the patch?

bobby285271 commented:
I resolved the merge conflict at  #167671 but haven't done any testing yet.
